scarlet/initialization.py

```diff
@@ -157,27 +189,71 @@ def trim_morphology(center_index, morph, bg_thresh):
                 bbox.stop[1] - center_index[-1],
             )
         )
-        while boxsize < size:
-            boxsize += 16  # keep box sizes quite small
+    else:
+        size = 0
 
-    # define bbox and trim to bbox
+    # define new box and cut morphology accordingly
+    boxsize = get_minimal_boxsize(size)
     bottom = center_index[0] - boxsize // 2
     top = center_index[0] + boxsize // 2
     left = center_index[1] - boxsize // 2
     right = center_index[1] + boxsize // 2
     bbox = Box.from_bounds((bottom, top), (left, right))
     morph = bbox.extract_from(morph)
-    bbox = Box.from_bounds((bottom, top), (left, right))
     return morph, bbox
 
 
+def init_compact_source(
+    sky_coord, frame, observations,
+):
+
+    # get PSF-corrected center pixel spectrum
+    spectrum = get_pixel_spectrum(sky_coord, observations, correct_psf=True)
+
+    # position in frame coordinates
+    center = frame.get_pixel(sky_coord)
+    center_index = np.round(center).astype(np.int)
+
+    # morphology initialized as a point source
+    morph_ = frame.psf.get_model().mean(axis=0)
+    origin = (
+        center_index[0] - (morph_.shape[0] // 2),
+        center_index[1] - (morph_.shape[1] // 2),
+    )
+    bbox_ = Box(morph_.shape, origin=origin)
+
+    # adjust box size to conform with extended sources
+    size = max(morph_.shape)
+    boxsize = get_minimal_boxsize(size)
+    morph = np.zeros((boxsize, boxsize))
+    origin = (
+        center_index[0] - (morph.shape[0] // 2),
+        center_index[1] - (morph.shape[1] // 2),
+    )
+    bbox = Box(morph.shape, origin=origin)
```

Comment:
This whole block seems like a lot of extra work, unless I'm missing something. Shouldn't `morph` just be the PSF image in a box shifted so that it is centered on the source location?

Reply:
Fair q. Our model psfs are *super* small (<~10 pixels), which is smaller than we allow any `ExtendedSource` to be. Hence the box size adjustment.